storage2/src/storage.rs

```diff
@@ -133,35 +136,33 @@ impl Storage {
                     self.0.write_node_batch(&batch.node_batch)?;
                     tracing::trace!(?jmt_root_hash, "wrote node batch to backing store");
 
-                    // Write the unwritten changes from the sidecar to RocksDB.
-                    for (k, v) in state.sidecar_changes.into_iter() {
-                        let sidecar_cf = db
-                            .cf_handle("sidecar")
-                            .expect("sidecar column family not found");
+                    // Write the unwritten changes from the nonconsensus to RocksDB.
+                    for (k, v) in state.nonconsensus_changes.into_iter() {
+                        let nonconsensus_cf = db
+                            .cf_handle("nonconsensus")
+                            .expect("nonconsensus column family not found");
 
                         match v {
-                            Some(v) => db.put_cf(sidecar_cf, k, &v)?,
+                            Some(v) => db.put_cf(nonconsensus_cf, k, &v)?,
                             None => {
-                                db.delete_cf(sidecar_cf, k)?;
+                                db.delete_cf(nonconsensus_cf, k)?;
                             }
                         };
                     }
 
                     // 4. update the snapshot
                     // Now that we've successfully written the new nodes, update the version.
                     let jmt_version = new_version;
-                    let snapshot = db.snapshot();
+                    let snapshot = Arc::new(db.snapshot());
                     // Obtain the write-lock for the latest snapshot, and replace it with the new snapshot.
                     let mut guard = self.0.latest_snapshot.write();
                     *guard = Arc::new(Snapshot::new(snapshot, jmt_version, db));
                     // Drop the write-lock (this will happen implicitly anyways, but it's good to be explicit).
                     drop(guard);
```

Comment:
I think you can do this like:

```rust
*self.0.latest_snapshot.write() = Arc::new(Snapshot::new(Arc::new(db.snapshot()), jmt_version, db));
```

Whether that's more readable is debatable, though :upside_down_face:

Reply:
Thinking about why we need the double-`Arc`: the outer one is because we may have many references to one of our `Snapshot`s, but why does `Snapshot` itself need to wrap the `rocksdb::Snapshot` in an `Arc`? Could we elide that? What gets in the way?

Reply:
Without having the `rocksdb::Snapshot` wrapped in an `Arc`, the tokio tasks that need to access it don't work out.

Reply:
Ah, that makes sense. I think we should leave it as is; an extra indirection is fine.

Reply:
> the outer one is because we may have many references to one of our `Snapshot`s

Why does that mean we need an `Arc`-wrapper? What if we just make `Snapshot` `Clone`?